util/artifacts.go

```diff
@@ -0,0 +1,95 @@
+package util
+
+import (
+	"context"
+	"errors"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/mongodb/grip"
+)
+
+// ArchiveContentFile represents a tar file on disk.
+type ArchiveContentFile struct {
+	Path string
+	Info os.FileInfo
+}
+
+func FindContentsToArchive(ctx context.Context, rootPath string, includes, excludes []string) ([]ArchiveContentFile, error) {
+	out := []ArchiveContentFile{}
+
+	files, errs := streamArchiveContents(ctx, rootPath, includes, excludes)
+	for fn := range files {
```

Comment:
I think this will deadlock if there's an error. `streamArchiveContents` will block on sending on the error channel, not closing the output channel, so this range will block.